src/main/java/com/questhelper/QuestHelperConfig.java

```diff
@@ -206,6 +208,78 @@ public static QuestFilter[] displayFilters()
 		}
 	}
 
+	enum SkillFilter implements Predicate<QuestHelper>
+	{
+		ANY(true),
+		ATTACK,
+		DEFENCE,
+		STRENGTH,
+		HITPOINTS,
+		RANGED,
+		PRAYER,
+		MAGIC,
+		COOKING,
+		WOODCUTTING,
+		FLETCHING,
+		FISHING,
+		FIREMAKING,
+		CRAFTING,
+		SMITHING,
+		MINING,
+		HERBLORE,
+		AGILITY,
+		THIEVING,
+		SLAYER,
+		FARMING,
+		RUNECRAFT,
+		HUNTER,
+		CONSTRUCTION;
+
+		private final Predicate<QuestHelper> predicate;
+
+		@Getter
+		private final String displayName;
+
+		SkillFilter()
+		{
+			this(false);
+		}
+
+		SkillFilter(boolean noPredicate)
+		{
+			if (!noPredicate)
+			{
+				this.predicate = q -> {
+					List<ExperienceReward> experienceRewards = q.getQuest().getQuestHelper().getExperienceRewards();
+					if (experienceRewards != null)
+					{
+						return experienceRewards.stream().anyMatch(reward -> reward.getSkill() == Skill.valueOf(this.toString()));
+					}
+					else
+					{
+						return false;
+					}
+				};
+			}
+			else
+			{
+				this.predicate = q -> true;
+			}
+			this.displayName = Text.titleCase(this);
+		}
+
+		@Override
+		public boolean test(QuestHelper quest)
+		{
+			return predicate.test(quest);
```

Comment:
nit: consistent with `this.` prefix
```suggestion
			return this.predicate.test(quest);
```